sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -379,6 +381,33 @@ class Analyzer(catalog: Catalog,
     }
   }
 
+  object ResolveSortAlias extends Rule[LogicalPlan] {
+    def apply(plan: LogicalPlan): LogicalPlan = plan transform {
+      case s @ Sort(ordering, global, a @ Aggregate(grouping, aggs, child))
+          if !s.resolved && a.resolved =>
+        val missingInAggs = new ArrayBuffer[NamedExpression]()
+        val aliasedOrdering = ordering.zipWithIndex.map{
+          case (o, i) => {
+            o.child match {
+              case aggOrSub @ (_: AggregateExpression | _: Substring) => {
```

Comment:
I have tested substring in newest version of spark.
```
SELECT value, sum(key)
FROM src
GROUP BY value
ORDER BY substr(value, 1, 5)
```
is right.

```
SELECT substr(value, 1, 5), sum(key)
FROM src
GROUP BY substr(value, 1, 5)
ORDER BY substr(value, 1, 5)
```
is wrong, and ERROR LOG is `cannot resolve 'value' given input columns _c0, _c1; line 4 pos 16`.